packages/block-editor/src/components/block-title/index.js

```diff
@@ -26,55 +13,18 @@ import { store as blockEditorStore } from '../../store';
  * @example
  *
  * ```jsx
- * <BlockTitle clientId="afd1cb17-2c08-4e7a-91be-007ba7ddc3a1" />
+ * <BlockTitle clientId="afd1cb17-2c08-4e7a-91be-007ba7ddc3a1" maximumLength={ 17 }/>
  * ```
  *
  * @param {Object} props
  * @param {string} props.clientId Client ID of block.
+ * @param {number} props.maximumLength The maximum length that the block title string may be before truncated.
  *
- * @return {?string} Block title.
+ * @return {JSX.Element} Block title.
  */
-export default function BlockTitle( { clientId } ) {
-	const { attributes, name, reusableBlockTitle } = useSelect(
-		( select ) => {
-			if ( ! clientId ) {
-				return {};
-			}
-			const {
-				getBlockName,
-				getBlockAttributes,
-				__experimentalGetReusableBlockTitle,
-			} = select( blockEditorStore );
-			const blockName = getBlockName( clientId );
-			if ( ! blockName ) {
-				return {};
-			}
-			const isReusable = isReusableBlock( getBlockType( blockName ) );
-			return {
-				attributes: getBlockAttributes( clientId ),
-				name: blockName,
-				reusableBlockTitle:
-					isReusable &&
-					__experimentalGetReusableBlockTitle(
-						getBlockAttributes( clientId ).ref
-					),
-			};
-		},
-		[ clientId ]
-	);
-
-	const blockInformation = useBlockDisplayInformation( clientId );
-	if ( ! name || ! blockInformation ) return null;
-	const blockType = getBlockType( name );
-	const blockLabel = blockType
-		? getBlockLabel( blockType, attributes )
-		: null;
-	const label = reusableBlockTitle || blockLabel;
-	// Label will fallback to the title if no label is defined for the current
-	// label context. If the label is defined we prioritize it over possible
-	// possible block variation title match.
-	if ( label && label !== blockType.title ) {
-		return truncate( label, { length: 35 } );
-	}
-	return blockInformation.title;
+export default function BlockTitle( {
+	clientId,
+	maximumLength = MAXIMUM_TITLE_LENGTH,
+} ) {
+	return useBlockDisplayTitle( clientId, maximumLength );
```

Comment:
I think there's no need to specify a default here as `useBlockDisplayTitle` seems to already fall back to the default maximum length implicitly when `undefined` is passed:
```suggestion
export default function BlockTitle( {
	clientId,
	maximumLength,
} ) {
	return useBlockDisplayTitle( clientId, maximumLength );
```

That should hopefully mean it will be possible to remove the import of `MAXIMUM_TITLE_LENGTH`.